OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java

```diff
@@ -1141,7 +1145,11 @@ private RouteCalcResult searchRouteAndPrepareTurns(final RoutingContext ctx, Lat
 			ctx.finalRouteSegment = new BinaryRoutePlanner().searchRouteInternal(local, s, e, null);
 			result = RouteResultPreparation.convertFinalSegmentToResults(ctx, ctx.finalRouteSegment);
 			addPrecalculatedToResult(recalculationEnd, result);
-			makeStartEndPointsPrecise(result, s.getPreciseLatLon(), e.getPreciseLatLon());
+			if (ctx.calculationProgress.totalApproximateDistance == 0) {
```

Comment:
Previous code is cleaner and doesn't rely on strange calculationProgress

Reply:
70b15bc592